core/shared/utils/bh_hashmap.c

```diff
@@ -127,8 +127,13 @@ bh_hash_map_find(HashMap *map, void *key)
     HashMapElem *elem;
     void *value;
 
-    if (!map || !key) {
-        LOG_ERROR("HashMap find elem failed: map or key is NULL.\n");
+    if (!map) {
```

Comment:
My understanding is that we should modify acquire_wait_info in wasm_shared_memory.c but not here, e.g. change to:
```C
static AtomicWaitInfo *
acquire_wait_info(void *address, bool create)
{
    AtomicWaitInfo *wait_info = NULL;
    bh_list_status ret;

    if (address)    => add check here
        wait_info = (AtomicWaitInfo *)bh_hash_map_find(wait_map, address);

    if (!create)
        return wait_info;
```
@xujuntwt95329, what's you opinion? Qi found that "In acquire_wait_info the key of hasn_map_find can be NULL thus there are many senseless error logs".

Reply:
Yes, I agree with @wenyongh, the caller should avoid passing NULL pointer to `bh_hash_map_find`